src/App.jsx

```diff
@@ -14,45 +17,87 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
+
+function getPrepareGoods(goods, sortField, reversed) {
+  const prepareGoods = [...goods];
+
+  if (sortField) {
+    prepareGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    prepareGoods.reverse();
+  }
+
+  return prepareGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
+  const visibleGoods = getPrepareGoods(goodsFromServer, sortField, reversed);
+
+  const sortButtons = [
+    { name: 'Sort alphabetically', class: 'is-info', change: SORT_FIELD_NAME },
+    { name: 'Sort by length', class: 'is-success', change: SORT_FIELD_LENGTH },
+  ];
+
+  const reset = () => {
+    setReversed(false);
+    setSortField('');
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        {sortButtons.map(button => (
+          <button
+            key={button.name}
+            type="button"
+            className={cn('button', `${button.class}`, { 'is-light': sortField !== button.change })}
+            onClick={() => setSortField(`${button.change}`)}
```

Comment:
move to an own handler
```
const onGoodClick = (good) => () => {
  setSortField(good)
}
```
```suggestion
            onClick={onGoodClick(button.change)}
```